app/src/main/java/com/humberto/tasky/agenda/data/task/TaskRepositoryImpl.kt

```diff
@@ -72,15 +104,41 @@ class TaskRepositoryImpl @Inject constructor(
         }
     }
 
-    override suspend fun syncPendingTasks() {
-        taskDao.getTasksPendingSync(
-            userId = localUserId ?: ""
-        ).map { taskPendingSyncEntity ->
-            safeCall {
-                agendaApi.createTask(taskPendingSyncEntity.task.toTask().toTaskRequest())
-            }.onSuccess {
-                taskDao.deleteTaskPendingSync(taskPendingSyncEntity.taskId)
-            }
+    private inline fun <reified T: ListenableWorker> enqueuePendingWorker(taskId: String) {
+        val request = OneTimeWorkRequestBuilder<T>()
+            .setInputData(
+                workDataOf("TASK_ID" to taskId)
+            )
+            .setInitialDelay(15, TimeUnit.MINUTES)
+            .setConstraints(Constraints(
+                requiredNetworkType = NetworkType.CONNECTED
+            ))
+            .build()
+
+        workManager.enqueue(request)
+    }
+
+    override suspend fun syncPendingUpdateTask(taskId: String) {
+        val pendingTaskEntity = taskDao.getTaskPendingSync(
+            userId = localUserId ?: "",
+            taskId = taskId
+        )
+        safeCall {
+            agendaApi.updateTask(pendingTaskEntity.task.toTask().toTaskRequest())
+        }.onSuccess {
+            taskDao.deleteTaskPendingSync(pendingTaskEntity.taskId)
+        }
+    }
+
+    override suspend fun syncPendingCreateTask(taskId: String) {
+        val pendingTaskEntity = taskDao.getTaskPendingSync(
+            userId = localUserId ?: "",
+            taskId = taskId
+        )
+        safeCall {
+            agendaApi.createTask(pendingTaskEntity.task.toTask().toTaskRequest())
+        }.onSuccess {
+            taskDao.deleteTaskPendingSync(pendingTaskEntity.taskId)
```

Comment:
Here, `Result. retry()` is never returned, so this worker can either succeed or fail. You'll need to return that result to tell `WorkManager` that this work needs to be retried based on the retry policy you've set. Here is a useful extension function to return a worker result based on the network error received from your API request:

```
fun DataError.toWorkerResult(): ListenableWorker.Result {
    return when(this) {
        DataError.Local.DISK_FULL -> ListenableWorker.Result.failure()
        DataError.Network.REQUEST_TIMEOUT -> ListenableWorker.Result.retry()
        DataError.Network.UNAUTHORIZED -> ListenableWorker.Result.retry()
        DataError.Network.CONFLICT -> ListenableWorker.Result.retry()
        DataError.Network.TOO_MANY_REQUESTS -> ListenableWorker.Result.retry()
        DataError.Network.NO_INTERNET -> ListenableWorker.Result.retry()
        DataError.Network.PAYLOAD_TOO_LARGE -> ListenableWorker.Result.failure()
        DataError.Network.SERVER_ERROR -> ListenableWorker.Result.retry()
        DataError.Network.SERIALIZATION -> ListenableWorker.Result.failure()
        DataError.Network.UNKNOWN -> ListenableWorker.Result.failure()
    }
}
```

You'll also need to check how often the work has been executed if you want to limit it to a certain amount of retries.

```
if(runAttemptCount >= 5) {
  return Result.failure()
}
```